packages/service-utils/src/node/kafka.ts

```diff
@@ -98,18 +112,13 @@ export class KafkaProducer {
   async send(
     topic: string,
     messages: Record<string, unknown>[],
-    /**
-     * Reference: https://kafka.js.org/docs/producing#producing-messages
-     */
-    options?: {
-      acks?: number;
-      timeout?: number;
-      allowAutoTopicCreation?: boolean;
-    },
+    options?: KafkaProducerSendOptions,
   ): Promise<void> {
     if (!this.producer) {
       this.producer = this.kafka.producer({
         allowAutoTopicCreation: options?.allowAutoTopicCreation ?? false,
+        maxInFlightRequests: options?.maxInFlightRequests ?? 2000,
```

Comment:
Kafkajs default is unlimited (meaning a server will try to send all requests immediately). This allows a buffer to control memory usage. Events that are backed up will be queued in memory.